packages/headless/src/features/facets/facet-set/facet-set-insight-analytics-actions.ts

```diff
@@ -0,0 +1,161 @@
+import {FacetSortCriterion} from './interfaces/request';
+import {RangeFacetSortCriterion} from '../range-facets/generic/interfaces/request';
+import {
+  validatePayload,
+  requiredNonEmptyString,
+} from '../../../utils/validate-payload';
+import {facetIdDefinition} from '../generic/facet-actions-validation';
+import {Value} from '@coveo/bueno';
+import {
+  AnalyticsType,
+  makeInsightAnalyticsAction,
+} from '../../analytics/analytics-utils';
+import {
+  buildFacetBaseMetadata,
+  getStateNeededForFacetMetadata,
+  buildFacetSelectionChangeMetadata,
+} from './facet-set-analytics-actions-utils';
+
+export const logFacetShowMore = (facetId: string) =>
```

Comment:
@aelfaleh  As we discussed it seems we need to duplicate the analytics actions in order to connect them to the Insight analytics client. However, I see that the product listing facet is using the standard search actions. Now I'm wondering why product listing can reuse the search analytics actions while insight can't.

Can someone confirm if we're doing things the right way?

Reply:
It's a shame we have to duplicate all this stuff. @aelfaleh @olamothe this is definitely the way to go?

Reply:
@nathanlb if we choose to create a client analytics per use-case, yes

Reply:
I'm missing some context before I can answer.

Are the analytics events that are to be sent by the insight panel supposed to go through the standard analytics service ? Using the... "old"... search event system ? Or is it supposed to do it differently ?

Commerce events in general (product listing included) are handled in a very different manner, so it might not be a good idea to copy what product listing feature is doing.

If the "old analytics system" is what the insight panel needs to use, then this could probably be done in a more intelligent manner with some refactoring, with much less duplication.

I'm currently pretty swamped but I can try to take a note to look into check-outing this branch and giving it a go...

Reply:
We want it to track analytics the same way the current insight panel does which is the old search system. injecting a client would be more efficient if it isn't too limiting. @aelfaleh have you seen how that could be done and how it compares to the approach in this PR?

Reply:
I'll try it